### PR TITLE
Cps/implement moment coefficient

### PR DESCRIPTION
**Pull request by LHoeffner**
New Python process computes moment acting on airfoil for potential flow solutions. Unit test has been added and (simple) example validated against Xfoil result.

### CODE REVIEW

File: applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import KratosMultiphysics
+import numpy as np
```
Comment (rubenzorrilla) [inline applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py]: Use Kratos array types. `numpy` doesn't come as a default package of Python so it is possible that some users lack it.

File: applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+        settings.ValidateAndAssignDefaults(default_parameters)
+
+        self.body_model_part = Model[settings["model_part_name"].GetString()]
+        self.velocity_infinity = [0,0,0]
+        self.velocity_infinity[0] = settings["velocity_infinity"][0].GetDouble()
+        self.velocity_infinity[1] = settings["velocity_infinity"][1].GetDouble()
+        self.velocity_infinity[2] = settings["velocity_infinity"][2].GetDouble()
+        self.reference_area = settings["reference_area"].GetDouble()
+        self.reference_point = np.array([0.0,0.0,0.0])
```
Comment (rubenzorrilla) [inline applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py]:
```suggestion
        self.reference_point = KratosMultiphysics.Vector(3)
```

File: applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        self.create_output_file = settings["create_output_file"].GetBool()
+        self.reference_point[0] = settings["reference_point"][0].GetDouble()
+        self.reference_point[1] = settings["reference_point"][1].GetDouble()
+        self.reference_point[2] = settings["reference_point"][2].GetDouble()
+
+    def ExecuteFinalizeSolutionStep(self, refPoint = [0,0,0]):
+        print('COMPUTE MOMENT')
```
Comment (rubenzorrilla) [inline applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py]: Use the Kratos Logger for information output.

File: applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+        m = [0.0,0.0,0.0]
+
+        for cond in self.body_model_part.Conditions:
+            n =  np.array(cond.GetValue(KratosMultiphysics.NORMAL)) #Inigo: normal direction? (ass. outward of domain)
```
Comment (rubenzorrilla) [inline applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py]: Same `numpy` comment. And please remove comments that are not informative.

File: applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            cp = cond.GetValue(KratosMultiphysics.PRESSURE)
+            mid_point = cond.GetGeometry().Center()
+            mid_point =  np.array([mid_point.X, mid_point.Y, mid_point.Z])
+            lever = mid_point-self.reference_point
+            m += np.cross(lever, (-n*cp))
+
+        Cm = m[2]/self.reference_area
+
+        print('moment = ', m[2])
```
Comment (rubenzorrilla) [inline applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py]: Output using Kratos logger.

**Review by rubenzorrilla — CHANGES_REQUESTED** [review]

File: applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import KratosMultiphysics
+import numpy as np
+
+def Factory(settings, Model):
+    if( not isinstance(settings,KratosMultiphysics.Parameters) ):
```
Comment (philbucher) [inline applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py]:
```suggestion
    if not isinstance(settings,KratosMultiphysics.Parameters):
```

File: applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+                "create_output_file": false
+            }  """)
+
+        settings.ValidateAndAssignDefaults(default_parameters)
+
+        self.body_model_part = Model[settings["model_part_name"].GetString()]
+        self.velocity_infinity = [0,0,0]
+        self.velocity_infinity[0] = settings["velocity_infinity"][0].GetDouble()
+        self.velocity_infinity[1] = settings["velocity_infinity"][1].GetDouble()
+        self.velocity_infinity[2] = settings["velocity_infinity"][2].GetDouble()
```
Comment (philbucher) [inline applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py]:
```suggestion
        self.velocity_infinity = settings["velocity_infinity"].GetVector()
```

File: applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        self.reference_area = settings["reference_area"].GetDouble()
+        self.reference_point = np.array([0.0,0.0,0.0])
+        self.create_output_file = settings["create_output_file"].GetBool()
+        self.reference_point[0] = settings["reference_point"][0].GetDouble()
+        self.reference_point[1] = settings["reference_point"][1].GetDouble()
+        self.reference_point[2] = settings["reference_point"][2].GetDouble()
```
Comment (philbucher) [inline applications/CompressiblePotentialFlowApplication/python_scripts/compute_moment_process.py]: same

**Review by rubenzorrilla — APPROVED** [review]
Congratulations on your first PR!

**Pull request merged**